### DISCUSSION TITLE
Front End Composer Preview with a Custom Css

### DISCUSSION THREAD

**Discussion by Neo1975** [mia-platform/community#553] [Q&A] [answered]
### What product are you having troubles with?

Microfrontend Composer

### What Console version are you using?

13

### Description

Hi, 
I'm in the current situation: I developed a web component with angular and angular material with a custom theme.

### Actual Outcome

Custom Css are not correctly handled in preview

### Expected Outcome

Component work correctly but I have a couple of problem with style and css in preview.
1] body tag should have class mat-body anche ux-ng-material-theme. I modified Entry point in WebServer Configuration but in page composer preview the body inside the iframe missing that configuration
2] I put css theme in config map micro-lc-assets inside microservice micro-lc but is not imported in preview.

**Comment by epessina** [comment, marked as answer]
Hi @Neo1975 , thanks for reaching out!

Currently, the configurator preview has its own, separate, `index.html` and cannot inherit the entry point nor the styles mount as config maps.

Is the classes mandatory to be on the `<body>` tag? If not, you could assign them to the highest component in your page configuration. Similarly, you could mount a `<style>` tag containing the theme directly on your page (with the obvious drawback of repeating the theme for each page).

Anyhow, it may be beneficial to look into a more encapsulated way of handling your component's styling. I don't know if it can be applied to your use case, but it could be considered a good practice to have each web component carrying its styling, for example appending any necessary class on the ShadowRoot or using CSS variables with defaults for theming purposes.

Let me know if you need more details on the matter.

**Reply by Neo1975** [reply]
Hi Edoardo, thanks.
Is it possible to add a style tag with a hidden property depending on environment variable, I tried:
{
   "hidden":"{{rawObject eq ENVIRONMENT 'prod'}}"
}

but not working, can you help me?

**Reply by epessina** [reply]
You can use environment variables to interpolate portions of your JSON. For example, you can write 

```json
{
  "hidden": "{{SHOULD_HIDE}}"
}
```

and in deploy {{SHOULD_HIDE}} will be substituted with the appropriate value depending.

Having said that I'm not sure I understood what you are trying to achieve here, since a `<style>` tag would not understand a `hidden` property.

**Reply by Neo1975** [reply]
My aim is to produce a web component that can be hidden outside the preview mode but that help me to visualize correctly my components.
I tried your solution but it doesn't work, I think becouse **hidden** wants a **boolean**, in this way we pass a **string**. I tried also with "{{rawObject   SHOULD_HIDE}}" without success.

**Reply by epessina** [reply]
Environment variables can only be string. Also be aware that the interpolation feature at deploy time is not a back-kit feature, but a [Console feature](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/public_variables#interpolating-a-public-variable), so Handlebars-like helpers are not supported.

If `hidden` is a prop, isn't it something you can control in your web component then? Couldn't you accept a string and cast it to a boolean value?
